### PR TITLE
Hotfix/Fixing broken deploy pipeline on Azure DevOps

### PR DESCRIPTION
**Pull request by prathibhapadma**

### CODE REVIEW

File: scripts/pipelines/azure-devops/templates/deploy/deploy.sh [hunk 1]
```diff
@@ -1,10 +1,10 @@
 #!/bin/bash
 set -e
-# Add image name and ingress DNS name.
-export image="$2" dns="$3"
-yq eval '.spec.template.spec.containers[0].image = "'"$image"'"' -i "$4"
+# Add image name, dns_name and tag.
+source "$7"
```
[the remark this answers is not in the document]
Comment (prathibhapadma) [reply scripts/pipelines/azure-devops/templates/deploy/deploy.sh:4]: @sonatype-lift ignore

File: scripts/pipelines/azure-devops/templates/deploy/secrets.sh [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
-# export secrets name and add secrets into deployment file.
+# Deploy apps in exists namespace, If not exists Create new namespace.
+kubectl get namespace | grep -q "^$1" || kubectl create namespace "$1"
+# Command to create secrets to pull image from private registry.
 export secrets="$2" 
 yq e '.spec.template.spec."imagePullSecrets"=[{"name":"secrets"}]' -i "$6" 
 yq e '.spec.template.spec.imagePullSecrets[0].name = "'"$secrets"'"' -i "$6"
 # Apply the changes.
-kubectl apply -f "$6" --namespace="$1"
+kubectl create secret docker-registry "$2" --docker-server="$5" --docker-username="$3" --docker-password="$4" --namespace="$1" --save-config --dry-run=none -o yaml | kubectl apply -f "$6" --namespace="$1"
```
Comment (albertdb) [inline scripts/pipelines/azure-devops/templates/deploy/secrets.sh:10]:
```suggestion
kubectl create secret docker-registry "$2" --docker-server="$5" --docker-username="$3" --docker-password="$4" --namespace="$1" --save-config --dry-run=client -o yaml | kubectl apply -f -
```

File: scripts/pipelines/azure-devops/templates/deploy/deploy.sh [hunk 3]
```diff
@@ -1,10 +1,12 @@
 #!/bin/bash
 set -e
-# Add image name and ingress DNS name.
-export image="$2" dns="$3"
-yq eval '.spec.template.spec.containers[0].image = "'"$image"'"' -i "$4"
+# Add image name, dns_name and tag.
+# Run package-extra.sh to ${tag}
```
Comment (albertdb) [inline scripts/pipelines/azure-devops/templates/deploy/deploy.sh:4]:
```suggestion
# Source package-extra.sh to get ${tag}
```

File: scripts/pipelines/azure-devops/templates/deploy/deploy.sh [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
 yq eval '.spec.rules[0].host = "'"$dns"'"' -i "$5"
-# Create namespace in EKS cluster.
-kubectl create namespace "$1"
-# Apply manifest files.
-kubectl apply -f "$6" --namespace="$1"
\ No newline at end of file
+# Deploy apps in exists namespace, If not exists Create new namespace and apply manifest files.
+kubectl get namespace | grep -q "^$1" || kubectl create namespace "$1"
+kubectl apply -f "$6" --namespace="$1"
+kubectl rollout restart deploy --namespace="$1"
```
Comment (albertdb) [inline scripts/pipelines/azure-devops/templates/deploy/deploy.sh:12]:
```suggestion
# Create namespace if not exists 
kubectl get namespace | grep -q "^$1" || kubectl create namespace "$1"
# Apply manifest files
kubectl apply -f "$6" --namespace="$1"
# Rollout deployments in the namespace
kubectl rollout restart deploy --namespace="$1"
```

File: scripts/pipelines/azure-devops/templates/deploy/secrets.sh [hunk 5]
```diff
@@ -1,12 +1,10 @@
...
-# export secrets name and add secrets into deployment file.
+# Deploy apps in exists namespace, If not exists Create new namespace.
+kubectl get namespace | grep -q "^$1" || kubectl create namespace "$1"
+# Command to create secrets to pull image from private registry.
 export secrets="$2" 
 yq e '.spec.template.spec."imagePullSecrets"=[{"name":"secrets"}]' -i "$6" 
 yq e '.spec.template.spec.imagePullSecrets[0].name = "'"$secrets"'"' -i "$6"
 # Apply the changes.
-kubectl apply -f "$6" --namespace="$1"
+kubectl create secret docker-registry "$2" --docker-server="$5" --docker-username="$3" --docker-password="$4" --namespace="$1" --save-config --dry-run=client -o yaml | kubectl apply -f - --namespace="$1"
```
Comment (albertdb) [inline scripts/pipelines/azure-devops/templates/deploy/secrets.sh:10]:
```suggestion
# Create imagePullSecret if not exists
kubectl create secret docker-registry "$2" --docker-server="$5" --docker-username="$3" --docker-password="$4" --namespace="$1" --save-config --dry-run=client -o yaml | kubectl apply -f - --namespace="$1"
```

File: scripts/pipelines/azure-devops/templates/deploy/deploy.sh [hunk 6]
```diff
@@ -1,10 +1,22 @@
...
+# Add image name, dns_name and tag.
+# Source package-extra.sh to get ${tag}
+. "$7"
+
+# we get what is located after the last '/' in the branch name, so it removes /ref/head or /ref/head/<folder> if your branche is named correctly"
+branch_short=$(echo "$8" | awk -F '/' '{ print $NF }')
+
+# We change the name of the tag depending if it is a release or another branch
+echo "tag_completed: $8" | grep release && tag_completed="${tag}"
+echo "tag_completed_branch: $8" | grep release || tag_completed="${tag}_${branch_short}"
```
[the remark this answers is not in the document]
Comment (prathibhapadma) [reply scripts/pipelines/azure-devops/templates/deploy/deploy.sh:12]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (ultymatom) [reply scripts/pipelines/azure-devops/templates/deploy/deploy.sh:12]: @sonatype-lift unignore

SHould not be ignored, this value needs to be used

**Review by ultymatom — APPROVED** [review]

File: scripts/pipelines/azure-devops/templates/deploy/deploy.sh [hunk 7]
```diff
@@ -1,10 +1,22 @@
...
-yq eval '.spec.template.spec.containers[0].image = "'"$image"'"' -i "$4"
+# Add image name, dns_name and tag.
+# Source package-extra.sh to get ${tag}
+. "$7"
+
+# we get what is located after the last '/' in the branch name, so it removes /ref/head or /ref/head/<folder> if your branche is named correctly"
+branch_short=$(echo "$8" | awk -F '/' '{ print $NF }')
+
+# We change the name of the tag depending if it is a release or another branch
+echo "tag_completed: $8" | grep release && tag_completed="${tag}"
```
[the remark this answers is not in the document]
Comment (ultymatom) [reply scripts/pipelines/azure-devops/templates/deploy/deploy.sh:11]: @sonatype-lift ignore

File: scripts/pipelines/azure-devops/templates/deploy/deploy-pipeline.yml.template [hunk 8]
```diff
@@ -15,6 +15,7 @@ resources:
 
 variables:
   - group: registry_info 
+  - group: common-k8s-variables
```
Comment (albertdb) [inline scripts/pipelines/azure-devops/templates/deploy/deploy-pipeline.yml.template:18]:
```suggestion
  - group: deployment-env-variables
```

**Review by albertdb — APPROVED** [review]

**Pull request merged**